mlflow/pyfunc/__init__.py

```diff
@@ -438,26 +492,45 @@ def spark_udf(spark, model_uri, result_type="double"):
 
     def predict(*args):
         model = SparkModelCache.get_or_load(archive_path)
-        schema = {str(i): arg for i, arg in enumerate(args)}
+        input_schema = model.metadata.get_input_schema()
         pdf = None
+
+        def convert_type(series: pandas.Series, type_: np.dtype):
+            if series.dtype == type_:
+                return series
+            return series.astype(type_, errors="ignore")
```

Comment:
Should we consider using errors="coerce" here to convert invalid values into NaNs as described in [this SO post](https://stackoverflow.com/a/50858819)? Or actually, maybe we should just throw if the series cannot be converted into the expected type?

Reply:
That's a good question. I think both raising exception or conversion to NaNs are pretty invasive changes so if we want to do it we should do it before releasing this. Alternatively, we can add this behavior later protected by user controlled flag - e.g. setting --enforce_schema when loading/deploying model. 

I prefer the latter for the following reasons: 
a) We do not want to add barriers to starting using schemas - e.g. to avoid situation when adding schema breaks your workflow. 
b) If we ever record schema automatically (e.g. as part of autologging or autoextract it from tf models) we could break models that would otherwise work.